tests/permissions/setupCanForRules.test.js

```diff
@@ -0,0 +1,372 @@
+          helpline,
+          createdBy: workerSid,
+          accountSid,
+        },
+        user: new User('not creator', []),
+      },
+    ].map(t => ({ ...t, prettyConditionsSets: t.conditionsSets.map(arr => `[${arr.join(',')}]`) })),
+  ).test(
+    `Should return $expectedResult when $expectedDescription and conditionsSets are $prettyConditionsSets`,
+    async ({ action, conditionsSets, caseToBeCreated, user, expectedResult }) => {
+      const rules = buildRules(conditionsSets);
+      const can = setupCanForRules(rules);
+
+      const createdCase = await Case.create(caseToBeCreated, options);
+      expect(can(user, action, createdCase)).toBe(expectedResult);
+    },
+  );
+
+  // Test PostSurvey permissions
+  each(
+    [
+      {
+        action: getRandomPostSurveyAction(),
+        conditionsSets: [['everyone']],
+        expectedResult: true,
+        expectedDescription: 'not supervisor but set for everyone',
+        postSurveyToBeCreated: {
+          accountSid,
+          taskId: 'task-sid',
+          contactTaskId: 'contact-task-id',
+          data: {},
+        },
+        user: new User('not creator', []),
+      },
+      {
+        action: getRandomPostSurveyAction(),
+        conditionsSets: [],
+        expectedResult: false,
+        expectedDescription: 'not supervisor and set for no one',
+        postSurveyToBeCreated: {
+          accountSid,
+          taskId: 'task-sid',
+          contactTaskId: 'contact-task-id',
+          data: {},
+        },
+        user: new User('not creator', []),
+      },
+      {
+        action: getRandomPostSurveyAction(),
+        conditionsSets: [],
+        expectedResult: false,
+        expectedDescription: 'is supervisor but set for no one',
+        postSurveyToBeCreated: {
+          accountSid,
+          taskId: 'task-sid',
+          contactTaskId: 'contact-task-id',
+          data: {},
+        },
+        user: new User('not creator', ['supervisor']),
+      },
+      {
+        action: getRandomPostSurveyAction(),
+        conditionsSets: [['isSupervisor']],
+        expectedResult: true,
+        expectedDescription: 'is supervisor',
+        postSurveyToBeCreated: {
+          accountSid,
+          taskId: 'task-sid',
+          contactTaskId: 'contact-task-id',
+          data: {},
+        },
+        user: new User('not creator', ['supervisor']),
+      },
+    ].map(t => ({ ...t, prettyConditionsSets: t.conditionsSets.map(arr => `[${arr.join(',')}]`) })),
+  ).test(
+    `Should return $expectedResult when $expectedDescription and conditionsSets are $prettyConditionsSets`,
+    async ({ action, conditionsSets, postSurveyToBeCreated, user, expectedResult }) => {
+      const rules = buildRules(conditionsSets);
+      const can = setupCanForRules(rules);
+
+      const createdPostSurvey = await PostSurvey.create(postSurveyToBeCreated, options);
+      expect(can(user, action, createdPostSurvey)).toBe(expectedResult);
```

Comment:
I wonder if it's better to pass in the full set of possible actions and loop through all of them rather than just testing a random one each time? 

'fuzz testing' where you randomise the input is valuable when the range of possible inputs is very large, but if, like in this case, the range of possible values is small enough to exhaustively test without being slow, that's probably the best option